components/tidb_query_vec_executors/src/runner.rs

```diff
@@ -387,6 +391,9 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                 &mut ctx,
             )?;
 
+            // Check the deadline while finishing fetch data from executors.
+            self.deadline.check()?;
```

Comment:
In `self.internal_handle_request()`, [the deadline is checked](https://github.com/tikv/tikv/pull/8322/files#diff-9ec3d88bc161808120203a74bc3381b3R495), why do we still need to check it?